CakePHP/Sniffs/Commenting/TypeHintSniff.php

```diff
@@ -273,4 +287,47 @@ protected static function getValueNode(string $tagName, $tagComment): PhpDocTagV
 
         return $phpDocParser->parseTagValue(new TokenIterator($phpDocLexer->tokenize($tagComment)), $tagName);
     }
+
+    /**
+     * @param array<\PHPStan\PhpDocParser\Ast\Type\TypeNode> $types
+     * @return array<\PHPStan\PhpDocParser\Ast\Type\TypeNode>
+     */
+    protected function makeUnique(array $types): array
+    {
+        $typesAsString = [];
+
+        foreach ($types as $key => $type) {
+            $type = (string)$type;
+            if (in_array($type, $typesAsString, true)) {
+                unset($types[$key]);
+
+                continue;
+            }
+            $typesAsString[] = $type;
+        }
+
+        return $types;
+    }
+
+    /**
+     * @param array<\PHPStan\PhpDocParser\Ast\Type\TypeNode> $types
+     * @return bool
+     */
+    protected function isComplexObjectCollection(array $types, bool $isUnion): bool
+    {
+        if (!$isUnion) {
```

Comment:
Can't you just not call this function for non-union types?

Reply:
Probably, will have to refine a bit more